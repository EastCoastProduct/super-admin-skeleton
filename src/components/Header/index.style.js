import { StyleSheet } from 'aphrodite/no-important';
import variables from '../../styles/variables';

const { colors, sizes } = variables;

export default StyleSheet.create({
  header: {
    alignItems: 'center',
    backgroundColor: colors.header,
    color: colors.contentBcg,
    display: 'flex',
    height: sizes.header,
    justifyContent: 'flex-end',
    left: 0,
    padding: '0 50px',
    position: 'fixed',
    textTransform: 'uppercase',
    top: 0,
    width: '100%',
  },
  link: {
    color: 'inherit',
    marginLeft: 20,
  },
});
