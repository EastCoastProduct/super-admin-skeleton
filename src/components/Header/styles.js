import { StyleSheet } from 'aphrodite/no-important';
import { colors, sizes, zIndex } from '../../styles/variables';

export default StyleSheet.create({
  header: {
    alignItems: 'center',
    backgroundColor: colors.header,
    color: colors.contentBcg,
    display: 'flex',
    height: sizes.header,
    justifyContent: 'flex-end',
    left: 0,
    padding: '0 50px',
    position: 'fixed',
    textTransform: 'uppercase',
    top: 0,
    width: '100%',
    zIndex: zIndex.header,
  },
  link: {
    marginLeft: 20,
  },
});
