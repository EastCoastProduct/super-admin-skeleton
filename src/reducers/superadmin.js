import store from 'store';
import { fromJS } from 'immutable';
import { LOGIN_SUCCESS } from '../constants/actions';

const initialState = () => fromJS(store.get('superadmin')) || fromJS({});

export default (state = initialState(), action) => {
  switch (action.type) {
    case LOGIN_SUCCESS:
      return state.merge(fromJS(action.superadmin));
    default:
      return state;
  }
};
