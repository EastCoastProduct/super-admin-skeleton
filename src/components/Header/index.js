import React, { PropTypes } from 'react';
import { css } from 'aphrodite/no-important';
import styles from './index.style';

const Header = ({ handleLogout, superadmin }) =>
  <header className={css(styles.header)}>
    <p>Hello,&nbsp;
      <b>{superadmin.get('firstname')} {superadmin.get('lastname')}</b>
    </p>
    <a
      className={css(styles.link)}
      href
      onClick={handleLogout}
    >Logout</a>
  </header>;

Header.propTypes = {
  handleLogout: PropTypes.func.isRequired,
  superadmin: PropTypes.object.isRequired,
};

export default Header;
