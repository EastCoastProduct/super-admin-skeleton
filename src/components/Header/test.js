import React from 'react';
import { shallow } from 'enzyme';
import { fromJS } from 'immutable';
import { superadmin } from '../../fixtures/superadmin';
import Header from './';

describe('Header component', () => {
  const wrapper = shallow(
    <Header handleLogout={jest.fn()} superadmin={fromJS(superadmin)} />
  );
  const instance = wrapper.instance();

  describe('snapshot', () => {
    it('renders basic required data', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('renders with empty superadmin object', () => {
      const newWrapper = shallow(
        <Header handleLogout={() => {}} superadmin={fromJS({})} />
      );

      expect(newWrapper).toMatchSnapshot();
    });
  });

  describe('instance', () => {
    it('handle Logout link click', () => {
      wrapper.find('a').simulate('click');

      expect(instance.props.handleLogout).toHaveBeenCalled();
    });
  });
});
