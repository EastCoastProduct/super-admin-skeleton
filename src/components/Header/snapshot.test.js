import React from 'react';
import renderer from 'react-test-renderer';
import { fromJS } from 'immutable';
import { StyleSheetTestUtils } from 'aphrodite/no-important';
import { superadmin } from '../../fixtures/superadmin';
import Header from './';

describe('Header component snapshot', () => {
  beforeEach(() => {
    StyleSheetTestUtils.suppressStyleInjection();
  });
  afterEach(() => {
    StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
  });

  it('renders basic required data', () => {
    const tree = renderer.create(
      <Header handleLogout={() => {}} superadmin={fromJS(superadmin)} />
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders with empty superadmin object', () => {
    const tree = renderer.create(
      <Header handleLogout={() => {}} superadmin={fromJS({})} />
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
