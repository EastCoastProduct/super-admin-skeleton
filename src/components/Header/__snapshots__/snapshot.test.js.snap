exports[`Header component snapshot renders basic required data 1`] = `
<header
  className="header_uusume">
  <p>
    Hello,Â 
    <b>
      John
       
      Doe
    </b>
  </p>
  <a
    className="link_1lrrya9"
    href={true}
    onClick={[Function]}>
    Logout
  </a>
</header>
`;

exports[`Header component snapshot renders with empty superadmin object 1`] = `
<header
  className="header_uusume">
  <a
    className="link_1lrrya9"
    href={true}
    onClick={[Function]}>
    Logout
  </a>
</header>
`;
